modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/AndesKernelBoot.java

```diff
@@ -267,7 +277,14 @@ public static void startAndesStores() throws Exception {
         // directly wire the instance without wrapped instance
         messageStore = new FailureObservingMessageStore(createMessageStoreFromConfig(contextStoreInConfig));
         MessagingEngine messagingEngine = MessagingEngine.getInstance();
-        messagingEngine.initialise(messageStore, subscriptionEngine);
+        MessageExpiryManager messageExpiryManager = null;
+        //depends on the configuration bind the appropriate expiry manger with the messaging engine
+        if(isExpiryCheckEnabledInDLC){
+            messageExpiryManager = new DLCMessageExpiryManager(messageStore);
```

Comment:
I think we should not pass the 'messageStore' object. Instead we can use 'AndesContext.getInstance().getMessageStore()' inside the constructor. For this, we need to make sure we initialize the objects after the message store is set to the context.